pkg/apis/cluster/types.go

```diff
@@ -104,6 +105,65 @@ type ClusterSpec struct {
 	// any resource that does not tolerate the Taint.
 	// +optional
 	Taints []corev1.Taint
+
+	// The model list of resource modeling in this cluster. Each modeling name and quota can be customized by the user.
+	// If the user does not define the modeling name and modeling quota, it will be the default model.
+	// The default model index from 0 to 11. Each default model's cpu quota is (2^index-1, 2^index], index >= 0
+	// Each default model's memory quota is also (2^index-1, 2^index], index >= 0, for example, model0 is like this:
+	// - index: 0
+	// ranges:
+	//   - name: CPU
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: memory
+	//     min: 0 GB
+	//     max: 1 GB
+	// model10 is like this:
+	// - index: 10
+	// range:
+	//   - name: CPU
+	//     min: 512 C
+	//     max: 1024 C
+	//   - name: memory
+	//     min: 512 GB
+	//     max: 1024 GB
+	// model11 is like this:
+	// - index: 11
+	// range:
+	//   - name: CPU
+	//     min: 1024 C
+	//     max: MAXINT
+	//   - name: memory
+	//     min: 1024 GB
+	//     max: MAXINT
+	// +optional
+	ResourceModels []ResourceModel
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int
+
+	// Ranges describes the resource quota ranges.
+	// +optional
+	Ranges []ResourceModelItem
+}
+
+// ResourceModelItem describes the detail of each modeling quota that ranges from min to max.
+type ResourceModelItem struct {
+	// Name is the name for the resource that you want to categorize.
+	// +optional
+	Name string
```

Comment:
Name should be enum type

Reply:
Let's discuss this line. For now, this structure can support 4 kinds of resources. In the actual production environment, are there more resource names? If there is, I will customize an enum structure to support more resource names.

```go
const (
	// CPU, in cores. (500m = .5 cores)
	ResourceCPU ResourceName = "cpu"
	// Memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
	ResourceMemory ResourceName = "memory"
	// Volume size, in bytes (e,g. 5Gi = 5GiB = 5 * 1024 * 1024 * 1024)
	ResourceStorage ResourceName = "storage"
	// Local ephemeral storage, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
	// The resource name for ResourceEphemeralStorage is alpha and it can change across releases.
	ResourceEphemeralStorage ResourceName = "ephemeral-storage"
)
```